### ISSUE TITLE
[feature request] Make logging part configurable

### ISSUE THREAD

**Issue by jcyuan** [Speykious/SeeShark#44]
something like

public interface ILogSink
{
    bool IsEnabled(LogEventLevel level, string type);
    void Log(LogEventLevel level, string type, object? source, string messageTemplate);
    void Log(
      LogEventLevel level,
      string type,
      object? source,
      string messageTemplate,
      params object?[] propertyValues);
  }

and in SeeShark have a default console sink and then in the end user logic just implement a sink adapter and replace the default one to record logs.

**Comment by Speykious** [comment]
So it's about bringing customizability here?
https://github.com/Speykious/SeeShark/blob/cc4aa3b69053f08c92b77528ab95d326c91d7f72/SeeShark/FFmpeg/FFmpegManager.cs#L102-L126

**Comment by jcyuan** [comment]
> So it's about bringing customizability here?
> 
> https://github.com/Speykious/SeeShark/blob/cc4aa3b69053f08c92b77528ab95d326c91d7f72/SeeShark/FFmpeg/FFmpegManager.cs#L102-L126

yes please, because i need to record informations into log files.
not sure why `public interface ILogSink` this line was missed in my code above. just fixed.

**Comment by Speykious** [comment]
Ok, I'll see what I can do. Ideally you'll just need to setup a delegate, no need to create an interface.

**Comment by jcyuan** [comment]
> Ok, I'll see what I can do. Ideally you'll just need to setup a delegate, no need to create an interface.

ok~ it works too, but not sure which way will be more efficient, but i think it's ignorable.

**Comment by jcyuan** [comment]
ah, right, another is, can you make the paths configurable too?

https://github.com/Speykious/SeeShark/blob/cc4aa3b69053f08c92b77528ab95d326c91d7f72/SeeShark/Device/VideoDeviceManager.cs#L52-L55

**Comment by Speykious** [comment]
Not sure what you mean, path to what?
If it's for the logging file, then you're gonna specify it yourself anyway, no problem :p

**Comment by jcyuan** [comment]
i'm sorry, it's a public static method. my bad.😅

**Comment by jcyuan** [comment]
> Not sure what you mean, path to what? If it's for the logging file, then you're gonna specify it yourself anyway, no problem :p

no no, i meant, i thought `FFmpegManager.SetupFFmpeg` is internal use only, i found it's in `VideoManager` class, and i wanna pass my path into it.
but i just found `FFmpegManager.SetupFFmpeg` is a static public method, so i can just call it when my app booting up.
no worry, this is not regarding to the logging problem.
sorry just ignore it.

**Comment by jcyuan** [comment]
do these libs any special? or just download from ffmpeg official site it will just work fine?

**Comment by Speykious** [comment]
SeeShark 4 relies on the FFmpeg 5.0 LibAV* native libraries. In any case, if the versions aren't correct, it's going to tell you.
So yeah you can just download them from the official website. But you need the `libav` dynamic libraries, not the `ffmpeg` executable.

**Comment by jcyuan** [comment]
> SeeShark 4 relies on the FFmpeg 5.0 LibAV* native libraries. In any case, if the versions aren't correct, it's going to tell you. So yeah you can just download them from the official website. But you need the `libav` dynamic libraries, not the `ffmpeg` executable.

nice one, thanks so much ❤️
